src/silx/gui/plot/ImageStack.py

```diff
@@ -145,49 +147,90 @@ def _urlChanged(self, value):
         self.sigCurrentUrlIndexChanged.emit(value)
 
 
-class UrlList(qt.QWidget):
+class UrlList(qt.QListWidget):
     """List of URLs the user to select an URL"""
 
     sigCurrentUrlChanged = qt.Signal(str)
-    """Signal emitted when the active/current url change"""
+    """Signal emitted when the active/current url change. If url == '' consider it as there is not data to be displayed"""
+
+    sigUrlsRemoved = qt.Signal(tuple)
+    """signal emit when some url have been removed from the URL list. Provided as a tuple of url as strings"""
 
     def __init__(self, parent=None):
-        super(UrlList, self).__init__(parent)
-        self.setLayout(qt.QVBoxLayout())
-        self.layout().setSpacing(0)
-        self.layout().setContentsMargins(0, 0, 0, 0)
-        self._listWidget = qt.QListWidget(parent=self)
-        self.layout().addWidget(self._listWidget)
+        super().__init__(parent)
+        self._editable = False
+        # are we in 'editable' mode: for now if true then we can remove some item from the list
+
+        # menu to be triggered when in edition from right-click
+        self._menu = qt.QMenu()
+        self._removeAction = qt.QAction(
+            text="Remove",
+            parent=self
+        )
+        self._removeAction.setShortcuts(
+            [
+                # qt.Qt.Key_Delete,
+                qt.QKeySequence.Delete,
+            ]
+        )
+        self._menu.addAction(self._removeAction)
 
         # connect signal / Slot
-        self._listWidget.currentItemChanged.connect(self._notifyCurrentUrlChanged)
-
-        # expose API
-        self.currentItem = self._listWidget.currentItem
+        self.currentItemChanged.connect(self._notifyCurrentUrlChanged)
+
+    def setEditable(self, editable: bool):
```

Comment:
It would be good to have a few docstrings, e.g. : 
```suggestion
    def setEditable(self, editable: bool):
    """Toggle whether the user can remove some URLs from the list"""
```